tests/test_repository.py

```diff
@@ -256,6 +256,110 @@ def test_add_commit_push(self):
 
         shutil.rmtree(WORKING_REPO_DIR)
 
+    def test_add_commit_push_non_blocking(self):
+        repo = Repository(
+            WORKING_REPO_DIR,
+            clone_from=self._repo_url,
+            use_auth_token=self._token,
+            git_user="ci",
+            git_email="ci@dummy.com",
+        )
+
+        # Create dummy files
+        # one is lfs-tracked, the other is not.
+        with open(os.path.join(WORKING_REPO_DIR, "dummy.txt"), "w") as f:
+            f.write("hello")
+        with open(os.path.join(WORKING_REPO_DIR, "model.bin"), "w") as f:
+            f.write("hello")
+
+        repo.git_add()
+        repo.git_commit()
+        try:
+            url, result = repo.git_push(blocking=False)
+        except subprocess.CalledProcessError as exc:
+            print(exc.stderr)
+            raise exc
```

Comment:
Why are we having a try/except here? This seems off for a test

Reply:
Indeed, I suspect it was in an old test which got copy-pasted around. Fixed that in the last commit.